### ISSUE TITLE
[Android] - Tapping TextInput unfocuses it

### ISSUE THREAD

**Issue by BenjiFarquhar** [flutter/flutter#143434]
### Steps to reproduce

This happens on my Pixel C emulator but not my Pixel 6 emulator. 

1. Have the TextInput displayed on the screen with **_nothing focused_** for at least 8 to 10 seconds.
2. Tap the TextInput.
3. You can repeat the process as many times as you like, and it still happens. 

Chuck a breakpoint on `FocusNode.unfocus` to see the stack trace. `"TextInputClient.onConnectionClosed"` is passed to `Future<dynamic> _handleTextInputInvocation(MethodCall methodCall)` in `TextInput` class from a platform channel. 

### Expected results

TextInput focuses and remains focused. 

### Actual results

TextInput may briefly focus, soft keyboard opens, TextInput however does not remain focused. 

### Code sample

<details open><summary>Code sample</summary>

```dart

import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Text Input Example'),
        ),
        body: const Center(
          child: Padding(
            padding: EdgeInsets.all(8.0),
            child: TextField(
              decoration: InputDecoration(
                border: OutlineInputBorder(),
                labelText: 'Enter something',
              ),
            ),
          ),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

Here I show the bug, the keyboard opens for a few seconds, then I tap the TextInput a second time and it finally focuses. 

[textinputbug2.webm](https://github.com/flutter/flutter/assets/26356162/702a4d81-15e6-475d-a9cb-1d4c62e43cec)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.16.8, on macOS 14.2.1 23C71 darwin-arm64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] IntelliJ IDEA Community Edition (version 2023.2.2)
[✓] VS Code (version 1.86.1)
[✓] Connected device (7 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by dam-ease** [comment]
Hello @BenjiFarquhar.
Thanks for filing this.
I can't seem to reproduce this on my end both on an emulator device (Pixel 3a) and on a real Android device.

> the soft keyboard opens for a few seconds

From the video shared, I can't see this behaviour, could this be an issue with where in the `TextField` onTap was received? 
Also, can you try this on a real Android device to see if this reproduces there as well?

**Comment by BenjiFarquhar** [comment]
Hi there, @dam-ease. Thanks for taking a look. 

> From the video shared, I can't see this behaviour

My video shows me waiting, then near the end, I tap the TextInput, and the soft keyboard goes up for a few seconds, but the TextInput remains unfocused; then I tap the TextInput a second time, and the TextInput focuses. 

Can you wait 30 seconds with nothing focused, then tap the TextInput on your Pixel C emulator?

> could this be an issue with where in the TextField onTap was received?

I don't understand what you mean.

**Comment by BenjiFarquhar** [comment]
@dam-ease No, in mine, the field becomes unfocused as soon as you tap it, but the keyboard opens. So, keyboard with no focus. 

Yours was the opposite; the field is focused without a keyboard opening. So, focus with no keyboard.

**Comment by dam-ease** [comment]
Thanks for checking this out.
I have tried producing this severally on my device but can only reproduce as above.
Can you help with the full device information you are testing with?

**Comment by BenjiFarquhar** [comment]
Cheers. This might be an emulator issue, so it should be closed. I have just started using a different emulator to work around it. But here are the specs if you want to reproduce the issue on this emulator. 

<img width="1387" alt="Screenshot 2024-02-19 at 19 08 17" src="https://github.com/flutter/flutter/assets/26356162/2f0d6f19-1b04-480e-981c-4f570200dc8b">
<img width="1085" alt="Screenshot 2024-02-19 at 19 07 55" src="https://github.com/flutter/flutter/assets/26356162/6154b294-f870-459c-994f-5b40aea642e6">
<img width="1376" alt="Screenshot 2024-02-19 at 19 07 45" src="https://github.com/flutter/flutter/assets/26356162/311d39b5-4702-4fa5-8a2e-7e15711c6450">

**Comment by dam-ease** [comment]
Thanks for sharing these. 
Looking through, this is quite similar to the device I ran my tests on as well but with different API levels, my device is a `Pixel C API 34`.
This might be specific to this emulator, I'll label this for more insights per what you shared in original comment:

> Chuck a breakpoint on FocusNode.unfocus to see the stack trace. "TextInputClient.onConnectionClosed" is passed to Future<dynamic> _handleTextInputInvocation(MethodCall methodCall) in TextInput class from a platform channel.

You could also share any logs that pop on your end to provide more helpful insights into the issue.

**Comment by justinmc** [comment]
CC @LongCatIsLooong from triage.
